tasks/data_handling/tests/test_data_pre_processing.py

```diff
@@ -0,0 +1,283 @@
+import scanpy as sc
+import unittest
+import os
+import numpy as np
+from collections import Counter
+
+import global_settings as gs
+from tasks.data_handling import data_pre_processing
+from tasks.test_helper.builders.data_handling import data_pre_process as test_data_pre_process
+from tasks.test_helper.builders.utils import context_utils as cu
+
+
+class TestCreateCountMatrix(unittest.TestCase):
+    def test_need_file_path(self):
+        with self.assertRaises(ValueError) as context:
+            result = data_pre_processing.create_count_matrix()
+
+    data_path = os.path.join('\\'.join(os.getcwd().split('\\')[:-2]), gs.TEST_DATA_PATH)
+
+    def tearDown(self) -> None:
+        cu.clean(folder_path=self.data_path)
+
+    file_name = 'simulated_csc_h5_data.h5'
+    simulated_matrix = test_data_pre_process.simulate_csc_h5_data(file_path=data_path,
+                                                                  filename=file_name)
+    simulated_adata = data_pre_processing.create_count_matrix(file_path=data_path + '\\' + file_name)
+
+    def test_equal_values_equal_index(self):
+        equal = True
+        indices = self.simulated_adata.X.nonzero()
+        rows = indices[0]
+        columns = indices[1]
+
+        for i in range(len(rows)):
+            row = rows[i]
+            col = columns[i]
+            if self.simulated_adata.X[row, col] != self.simulated_matrix.A[row, col]:
+                equal = False
+        self.assertTrue(equal)
+
+    def test_all_var_names_unique(self):
+        name_list = np.array(self.simulated_adata.var_names).tolist()
+        names_unique = len(name_list) == len(set(name_list))
+        self.assertTrue(names_unique)
+
+
+class TestQualityControl(unittest.TestCase):
+    data_path = os.path.join('\\'.join(os.getcwd().split('\\')[:-2]), gs.TEST_DATA_PATH)
+
+    def tearDown(self) -> None:
+        cu.clean(folder_path=self.data_path)
+
+    file_name = 'simulated_csc_h5_data.h5'
+    test_data_pre_process.simulate_csc_h5_data(file_path=data_path,
+                                               filename=file_name)
+    simulated_adata = data_pre_processing.create_count_matrix(file_path=data_path + '\\' + file_name)
+    data_pre_processing.quality_control(simulated_adata)
+    n_genes = simulated_adata.n_vars
+    n_cells = simulated_adata.n_obs
+
+    def test_n_genes_by_count(self):
+        cells = self.simulated_adata.X.nonzero()[0]
+        counted_values = Counter(cells)
+        equal = True
+        for cell, count in counted_values.items():
+            if count != self.simulated_adata.obs['n_genes_by_counts'].iloc[cell]:
```

Comment:
Remove element-wise comparison with loop. Make assertion in the right place (see comments before)